src/read.rs

```diff
@@ -310,6 +311,49 @@ impl<R: Read + io::Seek> ZipArchive<R> {
             comment: footer.zip_file_comment,
         })
     }
+    /// Extract a Zip archive into a directory.
+    ///
+    /// Malformed and malicious paths are rejected so that they cannot escape
```

Comment:
This should probably say "Malformed or potentially malicious paths".

Perhaps we should make reference to `name_as_child` docs to enumerate the types of paths that will be rejected.